lib/aggregate/aggregate_store.rb

```diff
@@ -71,15 +73,45 @@ def self.included(model_class)
       model_class.extend ClassMethods
     end
 
+    def save
+      defined?(super) &&
+        RailsUpgradeHelpers::Version.if_version(
+          rails_5: -> { set_saved_changes },
+          rails_6: -> { set_saved_changes }
+        )
```

Comment:
you're only checking is `super` is defined to call `set_saved_changes`, but then you're calling `super` immediately in the next line whether or not `super` is defined.

Is this `defined?` check really necessary then?

Side note, you're calling `set_saved_changes` _before_ we actually call `super` (i.e. do the saving). What happens in the event the save fails? Do we still want to be setting `saved_changes`?